### DISCUSSION TITLE
Google Calendar on Android not opening files

### DISCUSSION THREAD

**Discussion by danielcharrua** [spatie/icalendar-generator#112] [Bugs] [answered]
Hi guys, when opening the ICS on an Android device I get **'imposible to launch event'**

Any ideas?

I'm creating the event as follows:

```
$event = Event::create()
        ->name($_POST['name'])
        ->startsAt(new DateTime($_POST['starts_at'], new DateTimeZone('Europe/Madrid')))
        ->endsAt(new DateTime($_POST['ends_at'], new DateTimeZone('Europe/Madrid')))
        ->address($_POST['address'])
        ->image($_POST['image'])
        ->description($_POST['description']);

// Create calendar and download
$calendar = Calendar::create()->event($event);
echo $calendar->get();
```

See the calendar file that the program is generating:

```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:spatie/icalendar-generator
BEGIN:VTIMEZONE
TZID:Europe/Madrid
BEGIN:STANDARD
DTSTART:20221030T030000Z
TZOFFSETFROM:+0200
TZOFFSETTO:+0100
END:STANDARD
BEGIN:DAYLIGHT
DTSTART:20230326T020000Z
TZOFFSETFROM:+0100
TZOFFSETTO:+0200
END:DAYLIGHT
END:VTIMEZONE
BEGIN:VTIMEZONE
TZID:UTC
BEGIN:STANDARD
DTSTART:20220626T144530Z
TZOFFSETFROM:+0000
TZOFFSETTO:+0000
END:STANDARD
END:VTIMEZONE
BEGIN:VEVENT
UID:641c660a62023
DTSTAMP:20230323T144530Z
SUMMARY:SeedSigner 101
DESCRIPTION:https://2140meetups.com/meetup/6443/
LOCATION:Av. Novo Mesoiro\, 2\, La Coruña\, España
IMAGE;VALUE=URI:https://2140meetups.com/wp-content/uploads/2023/03/seedsign
 er-101.png
DTSTART;TZID=Europe/Madrid:20230330T193000
DTEND;TZID=Europe/Madrid:20230330T203000
END:VEVENT
END:VCALENDAR
```

**Comment by rubenvanassche** [comment]
The output seems valid: https://icalendar.org/validator.html#results

Sometimes calendar clients can be quite confused with valid output, searching the error on google / stack overflow might provide an answer.

**Reply by danielcharrua** [reply]
Hello @rubenvanassche, 

I have also checked the code in the package readme on 
https://github.com/spatie/calendar-links#generate-add-to-calendar-links-for-google-ical-and-other-calendar-systems

And Google Calendar on Android also rejects it, so I think is not something on my code only and maybe in the code being generated by the package. Can you please try the code on the readme? It fails to me on Android devices using Google Calendar.

Thank you.

```
BEGIN:VCALENDAR
VERSION:2.0
PRODID:spatie/icalendar-generator
NAME:Laracon online
X-WR-CALNAME:Laracon online
BEGIN:VEVENT
UID:5ef5c3f64cb2c
DTSTAMP;TZID=UTC:20200626T094630
SUMMARY:Creating calendar feeds
DTSTART:20190306T150000Z
DTEND:20190306T160000Z
DTSTAMP:20190419T135034Z
END:VEVENT
END:VCALENDAR
```

**Reply by rubenvanassche** [reply]
Have no android so cannot help there, since this example is also valid following RFC I think something is wrong at the Google calendar client on Android.

**Comment by danielcharrua** [comment, marked as answer]
Found the error, was encoding ute-8 with BOM. The output file now is readable also with Google Calendar on Android.

Thank you @rubenvanassche
